pkg/importer/dataStream.go

```diff
@@ -703,6 +714,45 @@ func copy(r io.Reader, out string, qemu bool, imageSize string) error {
 	return nil
 }
 
+//ValidateSpaceConstraint - validates wheather there is enough space in PVC for both qcow2 image and converted raw image
+//Assumes it is possible to retrieve info from qcow2 file by means of qemu-img utility
+//Returns failure if either of the following happens
+//1. info cannot be retrieved from qcow2 file
+//2. Either ActualSize or VirtualSize are 0 - not specifed
+//3. ActualSize and VirtualSize together exceed available PVC Space
+func ValidateSpaceConstraint(qcow2Image string, destDir string) (bool, error) {
+
+	//Verify there is enough space in pvc before conversion
+	info, err := qemuOperations.Info(qcow2Image)
+
+	if err != nil {
```

Comment:
qemuOperations.Info guarantees that if err != nil then info == nil and if err == nil then info != nil. So you don't need both checks.

Reply:
if I check only that info != null then i assume that method Info(..) does not return error otherwise. Currently, this is the case, but if the method info(..) is changed so that err can be returned even though info is not nill, then there can be a bug in ValidateSpaceInfo and there will be no clear indication for that.

Reply:
Fair enough.